src/Sylius/Bundle/ResourceBundle/Validator/WithinCollectionUniqueCodeValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Validator;
+
+use Sylius\Component\Resource\Model\CodeAwareInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class WithinCollectionUniqueCodeValidator extends ConstraintValidator
```

Comment:
Perhaps we can name it ``UniqueWithinCollectionConstraint`` and use property accessor to make the field that is supposed to be unique configurable? This way we could use that for zones for example and not only for codes.